cloud-gcp/src/test/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpCredentialConnectorTest.java

```diff
@@ -0,0 +1,266 @@
+        final AuthenticatedContext authContext = createAuthContext();
+        when(contextBuilder.contextInit(authContext.getCloudContext(), authContext, null, null, false)).thenReturn(context);
+        when(context.getProjectId()).thenReturn(null);
+        when(context.getServiceAccountId()).thenReturn("some service id");
+
+        final CloudCredentialStatus status = underTest.verify(authContext);
+
+        Assert.assertNotNull(
+                "The returned CloudCredentialStatus instance is null!",
+                status);
+        Assert.assertEquals(
+                "Invalid credential status has specified!",
+                CredentialStatus.FAILED,
+                status.getStatus());
+    }
+
+    /**
+     * Testing the GcpContext checking mechanism. If the inner created
+     * GcpContext does not contains a valid service account id then a FAILED
+     * status should come back.
+     */
+    @Test
+    public void testForFailedStatusBecauseMissingServiceAccId() {
+        final AuthenticatedContext authContext = createAuthContext();
+        when(contextBuilder.contextInit(authContext.getCloudContext(), authContext, null, null, false)).thenReturn(context);
+        when(context.getProjectId()).thenReturn("some id");
+        when(context.getServiceAccountId()).thenReturn(null);
+
+        final CloudCredentialStatus status = underTest.verify(authContext);
+
+        Assert.assertNotNull(
+                "The returned CloudCredentialStatus instance is null!",
+                status);
+        Assert.assertEquals(
+                "Invalid credential status has specified!",
+                CredentialStatus.FAILED,
+                status.getStatus());
+    }
+
+    /**
+     * Testing the GcpContext checking mechanism. If the inner created
+     * GcpContext does not contains a valid Compute instance then a FAILED
+     * status should come back.
+     */
+    @Test
+    public void testForFailedStatusBecauseMissingCompute() {
+        final AuthenticatedContext authContext = createAuthContext();
+        when(contextBuilder.contextInit(authContext.getCloudContext(), authContext, null, null, false)).thenReturn(context);
+        when(context.getProjectId()).thenReturn("some id");
+        when(context.getServiceAccountId()).thenReturn("some service id");
+        when(context.getCompute()).thenReturn(null);
+
+        final CloudCredentialStatus status = underTest.verify(authContext);
+
+        Assert.assertNotNull(
+                "The returned CloudCredentialStatus instance is null!",
+                status);
+        Assert.assertEquals(
+                "Invalid credential status has specified!",
+                CredentialStatus.FAILED,
+                status.getStatus());
+    }
+
+    /**
+     * If a null would be passed to the GcpCredentialConnector's verify
+     * method than an IllegalArgumentException would be thrown.
+     */
+    @Test(expected = IllegalArgumentException.class)
```

Comment:
`@Test.expected` is too common, and not possible to check what was the reason. Would be better to use `@Rule` to handle exception